src/transaction/Transaction.js

```diff
@@ -851,6 +851,86 @@ export default class Transaction extends Executable {
         return this;
     }
 
+    /**
+     * This method removes a signature from the transaction based on the public key or signature provided.
+     *
+     * @param {PublicKey} publicKey - The public key associated with the signature to remove.
+     * @param {Uint8Array} [signature] - The signature to remove. If omitted, all signatures for the given public key will be removed.
+     * @returns {Uint8Array[]} The removed signatures.
+     */
+    removeSignature(publicKey, signature) {
```

Comment:
I'm not sure if removing a singular signature would actually have a use case. Removing all signatures for a specific publicKey does. Can this function be simplified? Or is there a use case for removing a single signature in this manner? In either a multi-node or chunked transaction use case, removing a single signature without removing all signatures for a publicKey might be problematic.

Reply:
My original idea was that if a signature is used instead of a public key, the function could perhaps try and determine the public key tied to the signature and remove all signatures for that public key. But after thinking about it further, I think it is not the best approach, and it would be better to just rely on the public key.

Reply:
A single public key can only be associated with one signature at a time. The addSignature() method already enforces this by checking if a signature has already been added for that public key. Given this, I agree that we can safely rely on the public key alone when removing signatures. I initially included the signature parameter because the original method proposal included both a public key and a signature, which led me to believe there might be a specific use case in mind. However, I totally agree that it would be simpler and more effective to rely only on the public key.